rust/src/tests/link.rs

```diff
@@ -140,3 +143,18 @@ fn unchanged_ctime_fails(ctx: &mut SerializedTestContext, ft: FileType) {
             })
         });
 }
+
+// link/04.t
+assert_enoent_named_file!(link, either);
+
+// link/09.t
+crate::test_case! {
+    /// link returns ENOENT if the source file does not exist
+    enoent_source_not_exists
+}
+fn enoent_source_not_exists(ctx: &mut TestContext) {
+    let source = ctx.gen_path();
+    let dest = ctx.gen_path();
+
+    assert_eq!(link(&source, &dest), Err(Errno::ENOENT));
```

Comment:
Isn't this redundant with the first half of the "either" test case?

Reply:
There were two distinct cases in the original. I don't think they're exactly the same, but I could be wrong.